client/my-sites/plans/jetpack-plans/product-card/index.tsx

```diff
@@ -78,6 +79,16 @@ const ProductCard: React.FC< ProductCardProps > = ( {
 		getSiteAvailableProduct( state, siteId, item.productSlug )
 	);
 
+	const jetpackUpgradesLocked = purchases.reduce(
+		( carry, purchase ) => ( purchase.isLocked ? true : carry ),
+		false
+	);
+
+	const existingPurchaseIsIapPurchase = purchases.reduce(
+		( carry, purchase ) => ( purchase.isInAppPurchase ? true : carry ),
+		false
+	);
```

Comment:
Both conditions can be rewritten into a simpler form with `purchases.some( ... )`.

Reply:
Updated 😅